### PR TITLE
Quark PCB for Planck/Preonic

### PR DESCRIPTION
**Pull request by npspears**
Quark provides alternate bottom row layouts for both the OLKB Planck and Preonic

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by npspears** [comment]
Thanks for looking this over!

File: keyboards/quark/rules.mk [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
+UNICODE_ENABLE = yes 	    # Unicode
```
Comment (ridingqwerty) [inline keyboards/quark/rules.mk:22]:
```suggestion
  UNICODE_ENABLE = yes 	    # Unicode
```

File: keyboards/quark/keymaps/via/rules.mk.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+VIA_ENABLE = yes
```
Comment (ridingqwerty) [inline keyboards/quark/keymaps/via/rules.mk.txt:1]: This file has to have the `.txt` suffix removed.

**Review by ridingqwerty — CHANGES_REQUESTED** [review]
The via `rules.mk` file needs the `.txt` suffix removed.
You could also consider moving `#define TAPPING_TOGGLE 2` into the keyboard-level `config.h` to cut down on the number of files.

**Review by npspears — COMMENTED** [review]
.txt ending remove
place #define TAPPING_TOGGLE 2 into the keyboard-level config.h

**Review by npspears — COMMENTED** [review]
tapping toggle relocated, and removed txt at end of file done

File: keyboards/quark/config.h [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x7070
+#define PRODUCT_ID      0x5340
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    Nasp
+#define PRODUCT         QUARK
+#define DESCRIPTION     ortholinear PCB with alternate bottom rows
```
Comment (drashna) [inline keyboards/quark/config.h:27]: This value is no longer used.

```suggestion
```

File: keyboards/quark/keymaps/default_4x12_grid/keymap.c [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+  * | TAB       |   Q  |   W  |   E  |   R  |   T  |   Y  |   U  |   I  |   O  |   P  | DEL   |
+  * |------------+------+------+------+------+-------------+------+------+------+------+------|
+  * | CTRL&ESC  |   A  |   S  |   D  |   F  |   G  |   H  |   J  |   K  |   L  |   ;  |  "    |
+  * |------------+------+------+------+------+------|------+------+------+------+------+------|
+  * | SHIFT     |   Z  |   X  |   C  |   V  |   B  |   N  |   M  |   ,  |   .  |   /  | ENTER |
+  * |-------+-------+-------+-------+-------+-------+------+------+------+------+------+------|
+  * |    PRINT   | Ctrl | ALT  | GUI  |LOWER |SPACE |SPACE|RAISE | LEFT | DOWN |  UP  |RIGHT  |
+  * `-----------------------------------------------------------------------------------------'
+  */
+  [0] = LAYOUT_4x12_Grid( \
```
Comment (drashna) [inline keyboards/quark/keymaps/default_4x12_grid/keymap.c:38]:
```suggestion
  [0] = LAYOUT_ortho_4x12( \
```

File: keyboards/quark/keymaps/default_4x12_grid/keymap.c [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+  * |   ~    | !    |  @   |  #  |  $   |  %   |  | ^  |  &   |  *  |  (    |  )  | BCKSPC  |
+  * |------------+------+------+------+------+-------------+------+------+------+------+----|
+  * |   \    |    1  |   2  |   3  |   4  |   5  |   6  |   7  |   8  |   9  |   0  |   /   |
+  * |------------+------+------+------+------+------|------+------+------+------+------+----|
+  * |       |       |      |      |      |      |      |   +   |   =  |      |      |       |
+  * |-------+-------+-------+-------+-------+-------+------+------+------+------+------+----|
+  * |     |    | ESC  |CTRL-ALT-DEL|TASK|    |     |     |     |    |  '|'  |   `   |       |
+  * `---------------------------------------------------------------------------------------'
+  */
+  [1] = LAYOUT_4x12_Grid( \
```
Comment (drashna) [inline keyboards/quark/keymaps/default_4x12_grid/keymap.c:57]:
```suggestion
  [1] = LAYOUT_ortho_4x12( \
```

File: keyboards/quark/keymaps/default_4x12_grid/keymap.c [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+  * |       |      |      |  UP  |       |      |     |   _   |     |   [   |  ]    |      |
+  * |------------+------+------+------+------+-------------+------+------+------+------+---|
+  * |       |     | LEFT | DOWN  | RIGHT |     |      |   -  |      |   [   |   ]   |      |
+  * |------------+------+------+------+------+-----+-----+------+------+------+------+-----|
+  * |   F1  | F2 |  F3  |   F4  | F5    | F6  | F7   | F8   | F9    | F10  | F11   | F12   |
+  * |-------+-------+-------+-------+-------+-------+------+------+------+------+------+---|
+  * |RESET |    |      |       |       |     |      |      |       |      |       |        |
+  * `---------------------------------------------------------------------------------------'
+     */
+  [2] = LAYOUT_4x12_Grid( \
```
Comment (drashna) [inline keyboards/quark/keymaps/default_4x12_grid/keymap.c:76]:
```suggestion
  [2] = LAYOUT_ortho_4x12( \
```

File: keyboards/quark/keymaps/default_4x12_mit/config.h [hunk 7]
```diff
@@ -0,0 +1,2 @@
+
```
Comment (drashna) [inline keyboards/quark/keymaps/default_4x12_mit/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

As well as:
```c
#pragma once
```

File: keyboards/quark/keymaps/default_5x12_2x3/config.h [hunk 8]
```diff
@@ -0,0 +1,2 @@
+
```
Comment (drashna) [inline keyboards/quark/keymaps/default_5x12_2x3/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/quark/quark.h [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+) \
+{ \
+  { k000, k001, k002,  k003,  k004, k005,  k006,  k007, k008,  k009,  k010, k011 }, \
+  { k100, k101, k102,  k103,  k104, k105,  k106,  k107, k108,  k109,  k110, k111 }, \
+  { k200, k201, k202,  k203,  k204, k205,  k206,  k207, k208,  k209,  k210, k211 }, \
+  { k300, k301, k302,  k303,  k304, k305,  k306,  k307, k308,  k309,  k310, k311 }, \
+  { k400, k401, KC_NO, KC_NO, k402, KC_NO, KC_NO, k403, KC_NO, KC_NO, k404, k405 } \
+}
+
+#define LAYOUT_4x12_Grid( \
```
Comment (drashna) [inline keyboards/quark/quark.h:52]:
```suggestion
#define LAYOUT_ortho_4x12( \
```

File: keyboards/quark/rules.mk (hunk 1 above)
Comment (drashna) [reply keyboards/quark/rules.mk:22]:
```suggestion
UNICODE_ENABLE = yes        # Unicode
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/quark/rules.mk [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+
```
Comment (drashna) [inline keyboards/quark/rules.mk:23]:
```suggestion

LAYOUTS = ortho_4x12
```

**Review by npspears — COMMENTED** [review]
all review comments (so far) have been added

**Review by npspears — COMMENTED** [review]
fixed the broken info.json - sorry about that

**Review by drashna — APPROVED** [review]

**Review by ridingqwerty — APPROVED** [review]

**Comment by ridingqwerty** [comment]
Thanks!

**Pull request merged**